ScheduleAPI/Models/DateNowRetriever.cs

```diff
@@ -0,0 +1,9 @@
+﻿using System;
+
+namespace ScheduleAPI.Models
+{
+    public sealed class DateNowRetriever : IDateNowRetriever
```

Comment:
Hate to be a stickler for naming, but can we rename this all to 'DateTimeNow' it is more precise to what it actually returns. The only thing we are adding into this object over the System.DateTime object is the ability to inject Now and UtcNow values.

Reply:
See other comments on the interface. One of the discussion points I wanted to bring up with you is the necessity of a schedule having any concept outside of UTC. The conversion feels like a view/consumer concern where the schedule would always be UTC. We can discuss.

Reply:
yes that is correct, the schedule domain is only every concerned with UTC, the resource representation is the only place where conversion is a concern. I've placed that logic into a 'Builder' class who really isn't actually a builder, he is a converter. So I should probably rename that as well, but that is outside of the context of building an interface and wrapper implementation of DateTime, so that the Now or UtcNow property can be injected...